src/t_ringbuf.c

```diff
@@ -19,7 +19,7 @@ static void
 test_wraparound(void)
 {
 	const size_t n = 1000;
-	ringbuf_t *r = malloc(ringbuf_obj_size);
+	ringbuf_t *r = (ringbuf_t *)malloc(ringbuf_obj_size);
```

Comment:
Redundant type casts add unecessary verbosity. I am all for compilers doing strong static analysis / "linting" by default, but gcc/clang options used in Makefile (implicit `-Wincompatible-pointer-types`, `-Wcast-qual` and some others) generally do a good job, while some other options can generate too excessive warnings. I don't know if this is a VS10 default, but warning on implicit `void *` cast to struct pointer is too excessive.

Again, this is perhaps a coding style question, which I should write down in a separate document for a reference. I use a variation of Kernel Normal Form (KNF) from the BSD/UNIX world.

Reply:
The cast is necessary if the C files are compiled as C++ (as I am).

Reply:
C is not a subset of C++. The languages are not fully compatible and, in the recent years, they have been diverging even more.

I can only suggest to compile the C code using the C compiler and separately link the object file or use as a statically linked library. :)